src/d/actor/d_a_obj_gnndemotakie.cpp

```diff
@@ -7,40 +7,91 @@
 #include "d/actor/d_a_obj_gnndemotakie.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
+#include "d/res/res_gnndemotakie.h"
+
+const char daObjGnntakie_c::M_arcname[] = "Gnndemotakie";
 
 /* 00000078-00000098       .text solidHeapCB__15daObjGnntakie_cFP10fopAc_ac_c */
-void daObjGnntakie_c::solidHeapCB(fopAc_ac_c*) {
-    /* Nonmatching */
+BOOL daObjGnntakie_c::solidHeapCB(fopAc_ac_c* a_this) {
+    return ((daObjGnntakie_c*)a_this)->create_heap();
 }
 
 /* 00000098-000001F4       .text create_heap__15daObjGnntakie_cFv */
-void daObjGnntakie_c::create_heap() {
-    /* Nonmatching */
+BOOL daObjGnntakie_c::create_heap() {
+#if VERSION == VERSION_DEMO
+    J3DModelData* mdl_data;
+    J3DAnmTextureSRTKey* btk_data;
+    BOOL ret = FALSE;
+    mdl_data = (J3DModelData*)dComIfG_getObjectRes(M_arcname, GNNDEMOTAKIE_BDL_GNN_DEMO_TAKI_E);
+#else
+    BOOL ret = FALSE;
+    J3DModelData* mdl_data = static_cast<J3DModelData*>(dComIfG_getObjectRes(M_arcname, GNNDEMOTAKIE_BDL_GNN_DEMO_TAKI_E));
+#endif
+    JUT_ASSERT(0x5A, mdl_data != NULL);
+    if (mdl_data != NULL) {
+        mpModel = mDoExt_J3DModel__create(mdl_data, 0, 0x11020203);
+        if (mpModel != NULL) {
+#if VERSION == VERSION_DEMO
+            btk_data = (J3DAnmTextureSRTKey*)dComIfG_getObjectRes(M_arcname, GNNDEMOTAKIE_BTK_GNN_DEMO_TAKI_E);
+#else
+            J3DAnmTextureSRTKey* btk_data = static_cast<J3DAnmTextureSRTKey*>(dComIfG_getObjectRes(M_arcname, GNNDEMOTAKIE_BTK_GNN_DEMO_TAKI_E));
+#endif
+            JUT_ASSERT(0x61, btk_data != NULL);
+            if (btk_data != NULL) {
+                if (mpBtkAnm.init(mdl_data, btk_data, true, J3DFrameCtrl::EMode_LOOP, 1.0f, 0, -1, false, FALSE)) {
+                    ret = TRUE;
+                }
+            }
+        }
+    }
+    return ret;
 }
 
 /* 000001F4-00000300       .text _create__15daObjGnntakie_cFv */
 cPhs_State daObjGnntakie_c::_create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, daObjGnntakie_c);
+    cPhs_State state = dComIfG_resLoad(&mPhase, M_arcname);
+    if (state == cPhs_COMPLEATE_e) {
+        state = cPhs_ERROR_e;
+        if (fopAcM_entrySolidHeap(this, solidHeapCB, 0x0)) {
+            cullMtx = mpModel->getBaseTRMtx();
+            init_mtx();
+            mpBtkAnm.getFrameCtrl()->setFrame(mpBtkAnm.getStartFrame());
+            mpBtkAnm.setPlaySpeed(1.0);
+            state = cPhs_COMPLEATE_e;
+        }
+    }
+    return state;
 }
 
 /* 000003A4-000003D4       .text _delete__15daObjGnntakie_cFv */
 bool daObjGnntakie_c::_delete() {
-    /* Nonmatching */
+#if VERSION == VERSION_DEMO
+    dComIfG_deleteObjectRes(M_arcname);
+#else
+    dComIfG_resDelete(&mPhase, M_arcname);
+#endif
```

Comment:
We have a macro to handle this common version difference
```suggestion
    dComIfG_resDeleteDemo(&mPhase, M_arcname);
```